### PR TITLE
Decompiled sub_69A997

### PR DESCRIPTION
**Pull request by LRFLEW**
This function seems really weird. I keep thinking I must have made a translation mistake, but I've triple-checked everything that looks odd to me and didn't see any errors. I have a feeling this will be rewritten (or at least heavily improved), so the oddities of this code may not matter as much.

### CODE REVIEW

File: src/peep/peep.c [hunk 1]
```diff
@@ -7133,10 +7133,86 @@ uint8 sub_69A60A(rct_peep* peep){
...
+				if (z != path->base_height) continue;
+				if (footpath_element_is_wide(path)) continue;
+			}
+			
+			if ((!footpath_element_is_queue(path) ||
+				 RCT2_GLOBAL(0x00F1AEE1, uint8) != path->properties.path.ride_index) &&
+				(path->type & RCT2_GLOBAL(0x00F1AEE0, uint8))) continue;
+			
+		} while (map_element_is_last_for_tile(path++));
+		if (!found) return score;
```
Comment (LinusU) [inline src/peep/peep.c]: It seems like `found` always will be `false` here? Or am I missing something?

Comment (LRFLEW) [inline src/peep/peep.c]: Shoot, you're right, I missed a line. I think I had that changed and forgot to stage it.

File: src/peep/peep.c [hunk 2]
```diff
@@ -7133,10 +7133,88 @@ uint8 sub_69A60A(rct_peep* peep){
...
+		test_edge = bitscanforward(edges);
+		if (test_edge == -1) return score;
+		edges &= ~(1 << test_edge);
+		if (edges == 0) {
+			if (footpath_element_is_sloped(path) &&
+					footpath_element_get_slope_direction(path) == test_edge) {
+				z += 2;
+			}
+			++RCT2_GLOBAL(0x00F1AEDE, sint16);
+			continue;
```
Comment (duncanspumpkin) [inline src/peep/peep.c]: Why not make this return sub69a997 then you could get rid of the while true

Comment (LRFLEW) [inline src/peep/peep.c]: That makes sense. I was wracking my brain on how to properly handle this situation, as the recursive part that occurs later acts significantly different than this in a few key areas. However, doing it that way makes sense.

File: src/peep/peep.c [hunk 3]
```diff
@@ -7133,10 +7133,86 @@ uint8 sub_69A60A(rct_peep* peep){
...
-static int sub_69A997(sint16 x, sint16 y, uint8 z, int test_edge) {
-	int eax = x, ebx, ecx = y, edx = z, esi, edi = 0xFFFF, ebp = test_edge;
-	RCT2_CALLFUNC_X(0x0069A997, &eax, &ebx, &ecx, &edx, &esi, &edi, &ebp);
-	return edi;
+static int sub_69A997(sint16 x, sint16 y, uint8 z, uint8 counter, uint16 score, int test_edge) {
+	x += TileDirectionDelta[test_edge].x;
+	y += TileDirectionDelta[test_edge].y;
+	
+	if (--RCT2_GLOBAL(0x00F1AED4, sint32) < 0) return score;
+	if (++counter > 0xc8) return score;
```
Comment (duncanspumpkin) [inline src/peep/peep.c]: change this to the literal 200

File: src/peep/peep.c [hunk 4]
```diff
@@ -7133,10 +7133,86 @@ uint8 sub_69A60A(rct_peep* peep){
...
+		
+		uint8 slope_dir;
+		if (footpath_element_is_sloped(path) &&
+			(slope_dir = footpath_element_get_slope_direction(path)) != test_edge) {
+			slope_dir ^= 2;
+			if (slope_dir != test_edge) continue;
+			if (path->base_height + 2 != z) continue;
+		} else {
+			if (z != path->base_height) continue;
+			if (footpath_element_is_wide(path)) continue;
```
Comment (duncanspumpkin) [inline src/peep/peep.c]: is this the right way round?

Comment (LRFLEW) [inline src/peep/peep.c]: I think so. I remember checking on this a few times, but I may just be misunderstanding it.

Comment (duncanspumpkin) [inline src/peep/peep.c]: Ahh yes your right. I hate the zero flag checking for zero.

File: src/peep/peep.c [hunk 5]
```diff
@@ -7133,10 +7133,86 @@ uint8 sub_69A60A(rct_peep* peep){
...
+		return sub_69A997(x, y, z, counter, score, test_edge);
+	}
+	if (RCT2_GLOBAL(0x00F1AEDE, sint16) != 0) {
+		--RCT2_GLOBAL(0x00F1AEDC, sint8);
+	}
+	--RCT2_GLOBAL(0x00F1AEDC, sint8);
+	if (RCT2_GLOBAL(0x00F1AEDC, sint8) < 0) return score;
+	do {
+		edges &= ~(1 << test_edge);
+		int k = RCT2_GLOBAL(0x00F1AEDC, int);
```
Comment (duncanspumpkin) [inline src/peep/peep.c]: change the name of this to `saved_f1aedc` or something as I really despise single letter variables that aren't related to directions or maths.

**Comment by duncanspumpkin** [comment]
Looks good. Can you play about with the white space a bit to split up the function. Also I personally don't like to write `++counter < N` when it would be more legible over two lines but thats a minor quibble that I am not fussed about.

**Comment by LRFLEW** [comment]
This should be good to go now.

**Pull request merged**